src/backendMiddleware.ts

```diff
@@ -44,14 +43,18 @@ export class BackendMiddleware {
     await this.storageLib.initConnection()
   }
 
-  public async setIdentityWallet(
+  public async authenticateAndSetIdentityWallet(
```

Comment:
Somehow I don't like the `And` in the name of the function (thinking of separation of concerns) and I don't know if we really need this function at all, as it is not used that frequently.
Would be interesting what others are thinking about that.

Reply:
This is essentially `authAndCache` or something like that. It would be used anytime we would like to update a cache entry perhaps.

Reply:
I don't like the name very much either. The previous name `setIdentityWallet` was more misleading (looks like a setter, when in fact it authenticates and then creates a new identityWallet object and sets that).

I think maybe `createIdentityWallet` or maybe just `authenticate` is sufficient? IMHO cache usage shouldn't really "show" in the public API of `backendMiddleware`

Reply:
unresolving this because not sure if you see my comment!

Reply:
issue will be handled in separate PR